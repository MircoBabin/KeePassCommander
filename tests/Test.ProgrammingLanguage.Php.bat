@echo off
    setlocal
    call "%~dp0Utils\Utils.test-setup.bat" "%~f0"
    
    "%~dp0..\assets\php\php.exe" "%~dp0Test.ProgrammingLanguage.Php.php" "%KeePassCommandExe%" > "%KeePassCommand_TestOutput%"
    
    call "%~dp0Utils\Utils.diff.bat"
    if errorlevel 1 exit /b 1
    exit /b 0