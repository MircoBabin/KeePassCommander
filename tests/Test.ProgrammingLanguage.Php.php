<?php

if ($argc < 2) throw new \Exception('Provide full path to KeePassCommand.exe as first commandline parameter');

$KeePassCommand_exe = $argv[1];
$KeePassEntry_php = str_replace('/', '\\', __DIR__ . '/..\src\ProgrammingLanguagesConnectors\KeePassEntry.php');

// require KeePassEntry.php containing class KeePassEntry
require_once($KeePassEntry_php);

$entry = new \KeePassCommander\KeePassEntry('Sample Entry', [
    'KeePassCommandExe' => $KeePassCommand_exe,
    'FieldNames' => [ 'extra field 1', 'extra password 1' ],
    'AttachmentNames' => [ 'example_attachment.txt' ],
]);
    
echo json_encode($entry, JSON_HEX_TAG | JSON_HEX_AMP | JSON_HEX_APOS | JSON_HEX_QUOT);
