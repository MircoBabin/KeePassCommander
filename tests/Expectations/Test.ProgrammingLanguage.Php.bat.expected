{"title":"Sample Entry","username":"User Name","password":"Password","url":"https:\/\/keepass.info\/","urlscheme":"https","urlhost":"keepass.info","urlport":"443","urlpath":"\/","notes":"Notes \u00c9\u00e9n \u20ac\u00fbr\u00f2","fields":{"extra field 1":"Some extra data","extra password 1":"Some extra password"},"attachments":{"example_attachment.txt":"This is an attachment.\r\n\r\nWith three lines in it."}}